pinot-core/src/main/java/org/apache/pinot/server/access/AllowAllAccessFactory.java

```diff
@@ -16,13 +16,19 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.server.api.access;
+package org.apache.pinot.server.access;
```

Comment:
@apucher In `BaseServerStarter` we have 
```String accessControlFactoryClass = _serverConf.getProperty(Server.ACCESS_CONTROL_FACTORY_CLASS, Server.DEFAULT_ACCESS_CONTROL_FACTORY_CLASS);``` 
where I have changed `DEFAULT_ACCESS_CONTROL_FACTORY_CLASS` to `org.apache.pinot.server.access.AllowAllAccessFactory`, and left `ACCESS_CONTROL_FACTORY_CLASS` unchanged (`org.apache.pinot.server.api.access.AllowAllAccessFactory`).
So if the current installation is using default (not configuring `pinot.server.admin.access.control.factory.class`) or configuring it to customized class there should be no problem. 

IIUC, the only case that will cause problem is someone explicitly configured `pinot.server.admin.access.control.factory.class` to `org.apache.pinot.server.api.access.AllowAllAccessFactory`. Do you know any use case configured in this way?

Reply:
At least, I would specifically call this out in the PR's release notes.

I'm aware of at least one installation that explicitly sets AllowAll... as part of startup scripts.

Reply:
Sure, added in the release notes.

Reply:
why are we changing the package names?

Reply:
We moved the access control classes from pinot-server to pinot-core since they are used in QueryServer of pinot-core to enable broker-server channel authorization. There will be circular dependency if we don't change the package name.

Reply:
Resolved